data/TaskDAO.js

```diff
@@ -0,0 +1,70 @@
+const Task = require('./Task');
+// In-memory implementation
+class TaskDAO {
+    constructor() {
+        this.tasks = [];
+        this.maxID = 0;
+    }
+
+    find(criteria) {
+        if(!criteria || Object.keys(criteria).length === 0) return this.tasks;
+        else {
+            return this.tasks.filter(task => this.matchesCriteria(task, criteria));
+        }
+    }
+
+    findById(id) {
```

Comment:
I prefer this style, where the "d" is lower case. You should be consistent though the methods in the controller all have `fooID()` and here you have `fooId()`

also this method is `findById` and you use a `filter` on the array instead of a `find` - that causes you to use an `[0]` on a result in another place.